src/openqaoa-core/openqaoa/problems/vehiclerouting.py

```diff
@@ -89,6 +89,103 @@ def G(self, input_networkx_graph):
 
+    @staticmethod
+    def from_distance_matrix(**kwargs):
+        """
+        Creates a instance of the vehicle routing problem, from the distance 
+        matrix.
+
+        Parameters
+        ----------
+        **kwargs:
+            Required keyword arguments are:
+            matrix: list, array
+                The number of nodes (vertices) in the graph.
+            n_vehicles: int
+                Number of vehicles used in the problem
+            method: str
+                method for the inequality constraints ['slack', 'unbalanced'].
+                For the unbalaced method refer to https://arxiv.org/abs/2211.13914
+                For the slack method: https://en.wikipedia.org/wiki/Slack_variable
+            subtours: list[list]
+                Manually add the subtours to be avoided
+            seed: int
+                Seed for the random problems.
+
+        Returns
+        -------
+            A instance of the vehicle routing problem.
+        """
+        matrix = kwargs.get("matrix")
+        n_nodes = len(matrix)
+        n_vehicles = kwargs.get("n_vehicles", 2)
+        pos = kwargs.get("pos", [])
+        method = kwargs.get("method", "slack")
+        if method == "slack":
+            penalty = kwargs.get("penalty", 4)
+        elif method == "unbalanced":
+            penalty = kwargs.get("penalty", [4, 1, 1])
+        else:
+            raise ValueError(f"The method '{method}' is not valid.")
+        subtours = kwargs.get("subtours", -1)
+        G = nx.Graph()
+        G.add_nodes_from(range(n_nodes))
+        for i in range(n_nodes - 1):
+            for j in range(i + 1, n_nodes):
+                G.add_weighted_edges_from([(i, j, matrix[i][j])])
+        return VRP(
+            G, n_vehicles, pos=pos, subtours=subtours, method=method, penalty=penalty
+        )
+
+    @staticmethod
+    def from_coordinates(**kwargs):
+        """
+        Creates a instance of the vehicle routing problem, from the distance 
+        matrix.
+
+        Parameters
+        ----------
+        **kwargs:
+            Required keyword arguments are:
+            pos: list, array
+                The nodes x, y (vertices) in the graph.
+            n_vehicles: int
+                Number of vehicles used in the problem
+            method: str
+                method for the inequality constraints ['slack', 'unbalanced'].
+                For the unbalaced method refer to https://arxiv.org/abs/2211.13914
+                For the slack method: https://en.wikipedia.org/wiki/Slack_variable
+            subtours: list[list]
+                Manually add the subtours to be avoided
+            seed: int
+                Seed for the random problems.
+
+        Returns
+        -------
+            A instance of the vehicle routing problem.
+        """
+        pos = kwargs.get("pos")
+        n_nodes = len(pos)
+        n_vehicles = kwargs.get("n_vehicles", 2)
+        pos = kwargs.get("pos", [])
```

Comment:
Line167 repeats the same assignment of `pos = kwargs.get("pos")`. 
I believe that this reassignment, in line170, might be not needed, or could be merged with the previous one.

Reply:
sure!

Reply:
I think the second pos is the one that should not go.